### PR TITLE
Ccw.core.test and cleanup

### PR DESCRIPTION
**Pull request by arichiardi**
Hello Laurent,

I quickly tried to enable the tests again, but I have a problem, maybe it happened to you as well. Basically eclipse starts but then hangs waiting for something. I do not know what it can be as I double check the conf with [Egit's](https://git.eclipse.org/c/egit/egit.git/tree/org.eclipse.egit.ui.test/pom.xml).

I have also added a profile to skip the tests altogether:
```mvn -P skip-ui-tests clean install```

But it really bothers me, because I was able to perform the tests...and through SWTBot create a fake project...just once. Then I don't know what happened. If you have any suggestion it is more than welcome.

### CODE REVIEW

**Comment by arichiardi** [comment]
I would say this one is low low priority, I opened it just because it can be useful at some point. Still no success though, on my machine no test is ever executed, maybe I should try on another OS...

**Comment by arichiardi** [comment]
Laurent, the app is waiting for something here in ```CCWDropAdapterEarlyStartup``` when I test, do have any idea of what? 

    "Worker-4" prio=10 tid=0x00007f9a70003000 nid=0x3e61 in Object.wait() [0x00007f9a63883000]
    java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000000c33003a8> (a org.eclipse.swt.internal.Lock)
	at java.lang.Object.wait(Object.java:503)
	at org.eclipse.swt.internal.Lock.lock(Lock.java:34)
	- locked <0x00000000c33003a8> (a org.eclipse.swt.internal.Lock)
	at org.eclipse.swt.internal.gtk.OS.g_utf16_to_utf8(OS.java:3355)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:63)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:54)
	at org.eclipse.swt.dnd.Transfer.registerType(Transfer.java:138)
	at org.eclipse.swt.dnd.FileTransfer.<clinit>(FileTransfer.java:39)
	at ccw.CCWDropAdapterEarlyStartup.<init>(CCWDropAdapterEarlyStartup.java:72)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at java.lang.Class.newInstance(Class.java:379)
	at org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.createExecutableExtension(RegistryStrategyOSGI.java:184)
	at org.eclipse.core.internal.registry.ExtensionRegistry.createExecutableExtension(ExtensionRegistry.java:905)
	at org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:243)
	at org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:55)
	at org.eclipse.ui.internal.WorkbenchPlugin.createExtension(WorkbenchPlugin.java:274)
	at org.eclipse.ui.internal.EarlyStartupRunnable.getExecutableExtension(EarlyStartupRunnable.java:117)
	at org.eclipse.ui.internal.EarlyStartupRunnable.run(EarlyStartupRunnable.java:66)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.ui.internal.Workbench$55.run(Workbench.java:2555)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:53)

**Comment by arichiardi** [comment]
I found out what the problem was! Basically ```CCWDropAdapterEarlyStartup``` was caching ```FileTransfer``` and ```UrlTransfer```...but this was not good for the test/surefire environment...

Now it works!, are you ready to be astonished? This is the screenshot of a failing test!

**Comment by laurentpetit** [comment]
Great news!

**Comment by arichiardi** [comment]
We should make a call for SWTBot tests. I saw they can even be registered without writing code, amazing!  :)

**Comment by arichiardi** [comment]
I am going to rebase this one over origin/master

**Comment by laurentpetit** [comment]
OK, I'm ready to move this PR forward. Would you please rebase it first on master. I intend to test and code review it tomorrow?

**Comment by arichiardi** [comment]
Sure!

**Comment by laurentpetit** [comment]
Hello, 

out of curiosity: why did you close this one?

**Comment by arichiardi** [comment]
Ops...I did some cleaning today and I deleted this branch...I hope I can undo..

**Comment by arichiardi** [comment]
I will need some reflog work but I am going to undelete it. Note for self, choose branch names that are close to PR titles..

**Comment by arichiardi** [comment]
Oh easier , thanks GitHub :)

**Comment by laurentpetit** [comment]
ok, let's code review this one ...

File: ccw.core.test/Counterclockwise Product Tests.launch [hunk 1]
```diff
@@ -28,17 +32,563 @@
...
+<mapEntry key="org.eclipse.wst.sse.core/dom/adapter/notification/time/criteria" value="10"/>
+<mapEntry key="org.eclipse.wst.sse.core/filebuffers/leaks" value="true"/>
+<mapEntry key="org.eclipse.wst.sse.core/filebuffers/lifecycle" value="false"/>
+<mapEntry key="org.eclipse.wst.sse.core/filebuffers/modelmanagement" value="false"/>
+<mapEntry key="org.eclipse.wst.sse.core/format" value="false"/>
+<mapEntry key="org.eclipse.wst.sse.core/structureddocument" value="false"/>
+<mapEntry key="org.eclipse.wst.sse.core/structuredmodel/lifecycle" value="false"/>
+<mapEntry key="org.eclipse.wst.sse.core/structuredmodel/locks" value="true"/>
+<mapEntry key="org.eclipse.wst.sse.core/structuredmodel/modelmanager" value="false"/>
+<mapEntr
```
Comment (laurentpetit) [inline ccw.core.test/Counterclockwise Product Tests.launch]: Is this super extra big addition to the file necessary? Won't it be fragile if used on different environments?

Comment (arichiardi) [inline ccw.core.test/Counterclockwise Product Tests.launch]: Woa, no I don't think it is necessary...will fix

File: ccw.core/src/java/ccw/CCWDropAdapterEarlyStartup.java [hunk 2]
```diff
@@ -59,6 +59,16 @@
...
+/**
+ * Startup configuration that needs to be executed early in the lifecycle.<br/>
+ * <br/>
+ * Note:<br/>
+ * The two instance variables below have been removed as they were causing CCW 
+ * to hang during tests, they probably cause deadlock during SWT startup in the
+ * test/junit environment.<br/><br/>
+ * <code>private final FileTransfer fileTransfer = FileTransfer.getInstance();</code>
+ * <code>private final URLTransfer urlTransfer = URLTransfer.getInstance();</code>
+ */
```
Comment (laurentpetit) [inline ccw.core/src/java/ccw/CCWDropAdapterEarlyStartup.java]: Good. I think you can remove the Note

File: ccw.feature/feature.xml [hunk 3]
```diff
@@ -48,12 +46,23 @@
...
+      <import plugin="org.eclipse.e4.ui.bindings"/>
+      <import plugin="org.eclipse.e4.ui.di"/>
+      <import plugin="org.eclipse.e4.ui.model.workbench"/>
+      <import plugin="org.eclipse.e4.ui.services"/>
+      <import plugin="org.eclipse.e4.ui.widgets"/>
+      <import plugin="org.eclipse.e4.ui.workbench"/>
+      <import plugin="org.eclipse.core.runtime.compatibility"/>
+      <import plugin="org.eclipse.osgi.services"/>
+      <import plugin="org.eclipse.ui.trace"/>
+      <import feature="org.eclipse.rcp" version="4.3.2.v20140221-1700"/>
```
Comment (laurentpetit) [inline ccw.feature/feature.xml]: Is it necessary to precisely specify the version here?

Comment (arichiardi) [inline ccw.feature/feature.xml]: No, this was done by Eclipse, will remove

**Comment by laurentpetit** [comment]
What is the mvn command line to use? Doing `mvn clean package`, for instance, does trigger `ccw.core.test` but its running time is suspect (a couple seconds)?

**Comment by arichiardi** [comment]
ahem...I don't know if I ever called it through maven ```mvn clean integration-test``` perhaps?
I also found [this doc](http://www.slideshare.net/mickaelistria/integration-tests-for-rcp-made-easy-with-swtbot-and-tycho) right about now, but don't have time to read it now, will do it later...

**Comment by laurentpetit** [comment]
ok I'll take a look. Things were automated a year ago, but I don't remember
how it used to work...

2015-05-22 16:00 GMT+02:00 Andrea Richiardi <notifications@github.com>:

> ahem...I don't know if I ever called it through maven mvn clean
> integration-test perhaps?
> I also found this doc
> <http://www.slideshare.net/mickaelistria/integration-tests-for-rcp-made-easy-with-swtbot-and-tycho>
> right about now, but don't have time to read it now, will do it later...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laurentpetit/ccw/pull/750#issuecomment-104667405>.
>



-- 
Laurent Petit

**Comment by arichiardi** [comment]
Yes I did call it through maven and indeed is ```mvn clean integration-test``` but it should be bound to ```package``` as well...maybe is fast because we have just one test, you should see a CCW appearing on the screen...

**Comment by arichiardi** [comment]
There also was a way to execute tests on a different ```$DISPLAY``` on linux because the tested Eclipse needs to have full focus. I am trying to find it.

**Comment by laurentpetit** [comment]
I'm on OS X. Made some progress, been able to test via Eclipse UI. Red
flag, will investigate.

2015-05-22 16:10 GMT+02:00 Andrea Richiardi <notifications@github.com>:

> There also was a way to execute tests on a different $DISPLAY on linux
> because the tested Eclipse needs to have full focus. I am trying to find it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laurentpetit/ccw/pull/750#issuecomment-104669363>.
>



-- 
Laurent Petit

**Comment by laurentpetit** [comment]
Old Smoke Tests of mine are still working, isn't it cool? :-)

**Comment by laurentpetit** [comment]
ccw.core.ClojureTest aren't, issue with test-lein-launch-configuration

**Comment by arichiardi** [comment]
Yes it is :)

**Comment by arichiardi** [comment]
It would be great to have a SWTBot recorder Run Configuration set...next level s**t :))) !

**Comment by laurentpetit** [comment]
?

**Comment by arichiardi** [comment]
I added a link, it allows to record the tests and convert them to code...it is really great to have it so that you can reproduce bugs..I have never used it but it really looks [cool](https://wiki.eclipse.org/SWTBot/Generator#How_to_get_the_maximal_benefit_from_it).

[Long ibm doc](https://www.ibm.com/developerworks/community/blogs/c914709e-8097-4537-92ef-8982fc416138/entry/building_reusable_eclipse_ui_test_automation_framework_with_swtbot?lang=en)

**Comment by laurentpetit** [comment]
could you create a specific issue targetting this potential addition? I'd
like to revive tests without adding to much on the plate. Just working
smoke tests for a start.

2015-05-22 16:27 GMT+02:00 Andrea Richiardi <notifications@github.com>:

> I added a link, it allows to record the tests and convert them to
> code...it is really great to have it so that you can reproduce bugs..I have
> never used it but it really looks cool
> <https://wiki.eclipse.org/SWTBot/Generator#How_to_get_the_maximal_benefit_from_it>
> .
>
> Long ibm doc
> <https://www.ibm.com/developerworks/community/blogs/c914709e-8097-4537-92ef-8982fc416138/entry/building_reusable_eclipse_ui_test_automation_framework_with_swtbot?lang=en>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laurentpetit/ccw/pull/750#issuecomment-104673824>.
>



-- 
Laurent Petit

**Comment by arichiardi** [comment]
Sure!

**Comment by laurentpetit** [comment]
- `ccw.util-test` references `ccw.common` which does not exist?
- namespace `ccw.util-test` is named `ccw.util_test`, like the filename?

**Comment by arichiardi** [comment]
does not exists? oh my...let me check...

**Comment by laurentpetit** [comment]
I am close to being able to have valid tests again here. From Eclipse, it is OK. From maven, it's Work In Progress.

**Comment by arichiardi** [comment]
the name of ccw.util-test with underscore is wrong of course...

**Comment by arichiardi** [comment]
Mmm...my version of ccw.util_test.clj does not have any reference to ccw.common...

    (ns ccw.util-test
      (:use clojure.test
        ccw.util))

**Comment by arichiardi** [comment]
Have you rebased? because I remember having added something like ```ccw.common``` but I don't think in this PR,,,

**Comment by arichiardi** [comment]
Have a look at [this](https://github.com/laurentpetit/ccw/blob/master/ccw.core.test/src/clj/ccw/test_common.clj), but is on master...I fixed the wrong namespace...sorry for the mess..

**Comment by laurentpetit** [comment]
I have done lots of changes here, rebased the whole thing, etc. so I will probably close this PR and open a new one with my changes

**Comment by arichiardi** [comment]
Ok, drop me a line if you need my assistance with something..

**Comment by laurentpetit** [comment]
There will be testing involved for sure, as this kind of feature is very
platform dependent, to say the least!

Le mardi 26 mai 2015, Andrea Richiardi <notifications@github.com> a écrit :

> Ok, drop me a line if you need my assistance with something..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laurentpetit/ccw/pull/750#issuecomment-105433273>.
>


-- 
Laurent Petit

**Pull request closed without merging**